tiny_dnn/core/framework/tensor_utils.h

```diff
@@ -120,48 +55,62 @@ void binary_tensor_tensor_elementwise_operation(Tensor<TD, kDim> &dst,
   auto psrc2 = src2.host_begin();
 
   for_i(true, dst.size(), [pdst, psrc1, psrc2, &f](size_t i) {
-    pdst[i] = f(psrc1[i], psrc2[i]);
+    *std::next(pdst, i) = f(*std::next(psrc1, i), *std::next(psrc2, i));
   });
 }
 
-template <typename TD, typename TS, typename F, size_t kDim>
-void unary_tensor_elementwise_operation(Tensor<TD, kDim> &dst,
-                                        const Tensor<TS, kDim> &src,
+template <typename TD, typename TDS, typename TS, typename TSS, typename F>
+void unary_tensor_elementwise_operation(Tensor<TD, TDS> &dst,
+                                        const Tensor<TS, TSS> &src,
                                         F f) {
   dst.resize(src.shape());
 
   auto pdst = dst.host_begin();
   auto psrc = src.host_begin();
 
-  for_i(true, dst.size(), [pdst, psrc, &f](size_t i) { pdst[i] = f(psrc[i]); });
+  for_i(true, dst.size(), [pdst, psrc, &f](size_t i) {
+    *std::next(pdst, i) = f(*std::next(psrc, i));
+  });
 }
 
-template <typename TD, typename TS1, typename TS2, typename F, size_t kDim>
-void binary_tensor_scalar_operation(Tensor<TD, kDim> &dst,
-                                    const Tensor<TS1, kDim> &src1,
+template <typename TD,
+          typename TDS,
+          typename TS1,
+          typename TS1S,
+          typename TS2,
+          typename F>
+void binary_tensor_scalar_operation(Tensor<TD, TDS> &dst,
+                                    const Tensor<TS1, TS1S> &src1,
                                     TS2 src2,
                                     F f) {
   dst.resize(src1.shape());
```

Comment:
@Randl I checked locally that all `dst.resize` in `tiny_dnn/core/framework/tensor_utils.h` should be replaced by `dst.reshape` since you removed the method in the last commit.